org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/WorkspaceEncodingProvider.java

```diff
@@ -47,30 +48,27 @@
 
 	@Override
 	public String getEncoding(URI uri) {
-		if (workspace != null) {
-			if (uri != null) {
-				Iterator<Pair<IStorage, IProject>> storages = storage2UriMapper.getStorages(uri).iterator();
-				while (storages.hasNext()) {
-					Pair<IStorage, IProject> storage = storages.next();
-					if (storage.getFirst() instanceof IEncodedStorage) {
-						try {
-							return ((IEncodedStorage) storage.getFirst()).getCharset();
-						} catch (CoreException e) {
-							LOG.error("Error getting file encoding", e);
-						}
-					}
+		if (workspace != null && uri != null) {
+			Iterator<Pair<IStorage, IProject>> storages = storage2UriMapper.getStorages(uri).iterator();
+			while (storages.hasNext()) {
+				Pair<IStorage, IProject> storage = storages.next();
+				if (storage.getFirst() instanceof IEncodedStorage) {
 					try {
-						String result = storage.getSecond().getDefaultCharset(true);
-						return result;
+						return ((IEncodedStorage) storage.getFirst()).getCharset();
 					} catch (CoreException e) {
-						LOG.error("Error getting project's default encoding", e);
+						LOG.warn("Error getting file encoding", e);
```

Comment:
When the encoding could not be retrieved from the resource (which is unlikely to happen anyway) the encoding will fall back to runtime default. So the issue is not critical. I would consider this not as an error and lowered the message level therefore.